include/inexor/vulkan-renderer/exception.hpp

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <vulkan/vulkan_core.h>
+
+#include <stdexcept>
+
+namespace inexor::vulkan_renderer {
+
+/// @brief A custom exception class for handling Vulkan API exceptions.
+class Exception : public std::exception {
```

Comment:
I would structure the exceptions  a bit different.

1. Use an own base exception class, which all other our exception derive from.
2. Use more specific exceptions for specific use cases, like Vulkan API exceptions,.
3. Also derive from std::runtime_error, you get the string message member for free.
4. The Vulkan API Exception would have the additional member with VkResult
5. Make an own namespace/folder for the exceptions.

Reply:
I just realized I don't even need to define and override `what()` when inheriting from `std::runtime_error`. Nice.

Reply:
Done.